File: Data Structures & Algorithms/anagram-groups/submission-0.java
class Solution {
    public List<List<String>> groupAnagrams(String[] strs) {
        Map<String, List<String>> map = new HashMap<>();
        List<List<String>> result = new ArrayList<>();

        for (String s: strs) {
            char[] chars = s.toCharArray();
            Arrays.sort(chars);
            String sorted = new String(chars);
            if (!map.containsKey(sorted)) {
                map.put(sorted, new ArrayList<>());
            }
            map.get(sorted).add(s);
        }

        for (List<String> list: map.values()) {
            result.add(list);
        }

        return result;
    }
}

File: Data Structures & Algorithms/string-encode-and-decode/submission-0.py
class Solution:
    mapper = {}
    n = 0

    def encode(self, strs: List[str]) -> str:
        s = str(self.n)
        self.n += 1
        self.mapper[s] = strs

        return s

    def decode(self, s: str) -> List[str]:
        return self.mapper.get(s, [])
